src/mixings/logit_sb_mixing.cc

```diff
@@ -161,5 +164,8 @@ Eigen::VectorXd LogitSBMixing::get_weights(
   weights(num_components - 1) =
       1.0 - std::accumulate(weights.data(),
                             weights.data() + num_components - 1, 0.0);
+  if (log) {
```

Comment:
you could reverse the order of these computations, i.e. by default work in log scale, so that

```
std::vector<double> cumsum_log(num_components, 0.0);
for (int h = 1; h < num_components; h++) {
  cumsum_log[h] = cumsum_log[h -1] + std::log(1 - sigmoid(eta[h-1]));
}
for (int h=0; h < num_components - 1; h++) {
   logweights(h) = log(sigmoid(eta[h])) +  cumsum_log[h];
} 
```
Also note that setting weights(num_components - 1) = 1 - sum(), can be avoided by setting the last eta to 1.